### ISSUE TITLE
Classification Modeling issue - logistic regression model: Connector reply error: grpc::StatusCode::UNKNOWN: 'Exception iterating responses: n_features must be integral, got '2' (<class 'str'>).'

### ISSUE THREAD

**Issue by subash28kumar** [nabeel-oz/qlik-py-tools#30]
Hi Nabeel,

Hope you doing great, just tried to implement the actual sample data for Logistic regression model alone, but when tired fitting the data 
EXTENSION PyTools.sklearn_Fit(TEMP_SAMPLES{Model_Name, N_Features});

getting the below error"
The following error occurred:
Connector reply error: grpc::StatusCode::UNKNOWN: 'Exception iterating responses: n_features must be integral, got '2' (<class 'str'>).'

Not sure, what am missing here, the actual data set is this, just thought to check with you.

IRN | Paying_Regularly | Age | Gender | Marital_Status | Employment_Ind | Num_Dependents | Receive_Benefits_Ind | Receive_Alimony_Ind | Receive_Child_Support_Ind | Med_exp_ind | Incarceration_Ind
2098 | Yes | 35 | M | Single | Yes | 0 | Yes | No | Yes | No | 0
8765 | No | 46 | M | Single | Yes | 2 | Yes | No | Yes | Yes | 1
4536 | Yes | 27 | F | Single | Yes | 0 | No | No | Yes | No | 0
4219 | No | 65 | F | Single | No | 3 | No | No | Yes | Yes | 1
7699 | No | 54 | F | Married | No | 2 | No | Yes | Yes | No | 1
9123 | No | 44 | M | Married | No | 1 | No | Yes | No | No | 1
5647 | No | 32 | F | Single | No | 0 | Yes | Yes | Yes | No | 0
3189 | Yes | 65 | M | Married | Yes | 0 | Yes | Yes | No | No | 0
6123 | No | 49 | M | Married | No | 3 | No | No | No | No | 1
4912 | No | 37 | M | Married | No | 3 | No | No | No | No | 1
5812 | Yes | 43 | F | Married | Yes | 0 | Yes | No | Yes | No | 0
8213 | No | 58 | F | Married | No | 1 | No | No | Yes | Yes | 0
5678 | No | 29 | M | Single | No | 2 | Yes | Yes | No | No | 1
2987 | No | 36 | M | Single | No | 3 | No | No | Yes | Yes | 1
6219 | No | 47 | F | Married | No | 2 | No | Yes | Yes | No | 1
6231 | Yes | 35 | F | Single | Yes | 0 | No | No | No | No | 0
8812 | No | 54 | M | Married | No | 1 | No | No | No | No | 1
7034 | No | 58 | M | Single | No | 2 | No | No | No | Yes | 0
8518 | No | 42 | F | Single | No | 1 | No | No | Yes | No | 1
4745 | No | 31 | M | Single | No | 0 | No | Yes | No | No | 1

[Estimators]:
LOAD * INLINE 
[
Model Name;Estimator;Hyperparameters
HR-Attrition-LR;LogisticRegression;
](delimiter is ';');

[Scaler]:
LOAD
	[Scaler],
	[Null Value Handling],
	[Scale Hashed Features],
	[Hyperparameters] AS [Scaler.Hyperparameters];
LOAD * INLINE 
[
Scaler;Null Value Handling;Scale Hashed Features;Hyperparameters
StandardScaler;zeros;true;with_mean=True|bool, with_std=True|bool
](delimiter is ';');

[Feature Definitions]:
LOAD
    Name,
    Variable_Type,
    Data_Type,
    Feature_Strategy,
    Hash_Features
FROM [lib://Feature Definitions/SampleDataforPrediction_V1.1 (1).xlsx]
(ooxml, embedded labels, table is Feature_Definitions);

[Train-Test]:
LOAD
    IRN,
    "Paying_Regularly",
    "Age",
    Gender,
    Marital_Status,
    Employment_Ind,
    Num_Dependents,
    Receive_Benefits_Ind,
    Receive_Alimony_Ind,
    Receive_Child_Support_Ind,
    Med_exp_ind,
    Incarceration_Ind
FROM [lib://Feature Definitions/SampleDataforPrediction_V1.1 (1).xlsx]
(ooxml, embedded labels, table is LR_Classification_Results_1);

// Set up a variable for the scaler parameters
LET vScalerArgs = 'scaler=' & peek('Scaler', 0, 'Scaler') & ',' & 
					'missing=' & peek('Null Value Handling', 0, 'Scaler') & ',' & 
                    'scale_hashed=' & peek('Scale Hashed Features', 0, 'Scaler') & ',' & 
                    peek('Scaler.Hyperparameters', 0, 'Scaler');
                    

// Set up a variable for execution parameters
LET vExecutionArgs = 'overwrite=true,test_size=0.3,calculate_importances=true';

LET i = 0;

// Create a model for each estimator
For Each vModel in FieldValueList('Model Name')
    
    // Set up a variable for this estimator's parameters
    LET vEstimatorArgs = 'estimator=' & peek('Estimator', $(i), 'Estimators') & ',' & peek('Hyperparameters', $(i), 'Estimators');
    
    // Set up a temporary table for the model parameters
    [MODEL_INIT]:
    LOAD
        [Model Name] as Model_Name,
        '$(vEstimatorArgs)' as EstimatorArgs,
        '$(vScalerArgs)' as ScalerArgs,
        '$(vExecutionArgs)' as ExecutionArgs    
    RESIDENT Estimators
    WHERE [Model Name] = '$(vModel)';
    
    // Use the LOAD...EXTENSION syntax to call the Setup function
    [Result-Setup]:
    LOAD
        model_name,
        result,
        timestamp
    EXTENSION PyTools.sklearn_Setup(MODEL_INIT{Model_Name, EstimatorArgs, ScalerArgs, ExecutionArgs});
    
    [FEATURES]:
    LOAD
    	'$(vModel)' as Model_Name,
        Name,
        Variable_Type,
        Data_Type,
        Feature_Strategy,
        Hash_Features
    RESIDENT [Feature Definitions];
    
    // Use the LOAD...EXTENSION syntax to call the Set_Features function
    [Result-Setup]:
    LOAD
        model_name,
        result,
        timestamp
    EXTENSION PyTools.sklearn_Set_Features(FEATURES{Model_Name, Name, Variable_Type, Data_Type, Feature_Strategy, Hash_Features});

    Drop table MODEL_INIT, FEATURES;   
    
    LET i = $(i) + 1;
  
Next vModel;

[TEMP_TRAIN_TEST]:
LOAD
IRN &'|'&
Paying_Regularly &'|'&
Age &'|'&
Gender &'|'&
Marital_Status &'|'&
Employment_Ind &'|'&
Num_Dependents &'|'&
Receive_Benefits_Ind &'|'&
Receive_Alimony_Ind &'|'&
Receive_Child_Support_Ind &'|'&
Med_exp_ind &'|'&
Incarceration_Ind As N_Features
RESIDENT [Train-Test];

LET i = 0;

// Train and Test each model
For Each vModel in FieldValueList('Model Name')

    [TEMP_SAMPLES]:
    LOAD
    	'$(vModel)' as Model_Name,
        N_Features
    RESIDENT [TEMP_TRAIN_TEST];
    
    // Use the LOAD...EXTENSION syntax to call the Fit function
    [Result-Fit]:
    LOAD
        model_name,
        result as fit_result, 
        time_stamp as fit_timestamp, 
        score_result, 
        score
    EXTENSION PyTools.sklearn_Fit(TEMP_SAMPLES{Model_Name, N_Features});

**Comment by subash28kumar** [comment]
I am able to solve the issue, sorry for the inconvenience caused, closing the same
